### PR TITLE
Add Cloud Deploy Automation

### PR DESCRIPTION
**Pull request by ChrisGe4**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/16654

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_clouddeploy_automation` (beta)
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/clouddeploy/resource_clouddeploy_automation_test.go.erb [hunk 1]
```diff
@@ -0,0 +1,117 @@
+<% autogen_exception -%>
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/clouddeploy/resource_clouddeploy_automation_test.go.erb:1]: This file will need a version guard per https://googlecloudplatform.github.io/magic-modules/develop/test/ (under "Handwritten")

That'll resolve the destroyproducer error- it's currently (correctly) only present at beta+, along with the resource implementation.

**Review by rileykarson — COMMENTED** [review]
note: Responding to follow up with a YAQS, not taking on review when this leaves draft.

File: mmv1/products/clouddeploy/Automation.yaml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+              description: "Target labels."
+              default_from_api: true
+  - !ruby/object:Api::Type::Array
+    name: "rules"
+    description: "Required. List of Automation rules associated with the Automation resource. Must have at least one rule and limited to 250 rules per Delivery Pipeline. Note: the order of the rules here is not the same as the order of execution."
+    required: true
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::NestedObject
+          name: "promote_release_rule"
```
Comment (melinath) [inline mmv1/products/clouddeploy/Automation.yaml:148]:
```suggestion
          name: "promoteReleaseRule"
```

File: mmv1/products/clouddeploy/Automation.yaml [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+name: 'Automation'
+description: |
+  An `Automation` enables the automation of manually driven actions for a Delivery Pipeline, which includes Release promotion amongst Targets, Rollout repair and Rollout deployment strategy advancement.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Automate your deployment': 'https://cloud.google.com/deploy/docs/automation'
+  api: 'https://cloud.google.com/deploy/docs/api/reference/rest/v1/projects.locations.deliveryPipelines.automations'
+base_url: 'projects/{{project}}/locations/{{location}}/deliveryPipelines/{{delivery_pipeline}}/automations'
+self_link: 'projects/{{project}}/locations/{{location}}/deliveryPipelines/{{delivery_pipeline}}/automations/{{name}}'
+min_version: beta
```
Comment (shuyama1) [inline mmv1/products/clouddeploy/Automation.yaml:24]: any particular reason that we need this to be in the beta provider only? I see that it's using v1 version

Comment (ChrisGe4) [reply mmv1/products/clouddeploy/Automation.yaml:24]: It's because the automation feature is currently in public preview

**Review by shuyama1 — COMMENTED** [review]
Overall looks good. Only some small questions and comments. Thanks!

File: mmv1/products/clouddeploy/Automation.yaml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    required: true
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: "targets"
+        description: "Contains attributes about a target."
+        required: true
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: "id"
```
[the remark this answers is not in the document]
Comment (ChrisGe4) [reply mmv1/products/clouddeploy/Automation.yaml:135]: A target selector could have id or labels or both.

**Review by shuyama1 — APPROVED** [review]
Thank you!

**Pull request merged**